### PR TITLE
CSS.escape() sample

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @addyosmani @surma @beaufortfrancois 
CC: @mathiasbynens

This is an example for the new CSS.escape() function.

There's a deployed demo at https://jeffy.info/samples/css-escape/index.html

### CODE REVIEW

File: css-escape/index.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+<p>
+  There are certain strings that are valid as when used as an HTML element's <code>id</code> or
+  <code>class</code> attributes, but not valid as a CSS selector string. They include strings
+  that begin with numbers, contain spaces, or contain characters with special meaning in CSS.
+  Mathias Bynens's <a href="https://mathiasbynens.be/notes/css-escapes">explanation</a> delves into
+  the full details.
+</p>
+<p>
+  The <code>CSS.escape()</code> method provides a convenient way to escape a string so that conforms
+  to the CSS selector requirements.
```
Comment (mathiasbynens) [inline css-escape/index.html]: Maybe add something like this here:

> This is especially useful when dynamically building a CSS selector based on (untrusted) user input.

[Some examples of that from my CSS Day 2014 talk:](https://vimeo.com/100264064#t=20:25)
- `document.querySelector(location.hash)` should be `document.querySelector('#' + CSS.escape(location.hash.slice(1)))`
- `document.querySelector('a[href="' + someValue + '"]')` should be `document.querySelector('a[href="' + CSS.escape(someValue) + '"]')`

**Comment by mathiasbynens** [comment]
Thanks, @jeffposnick!

This sample demonstrated `CSS.escape` functionality, but doesn’t explain why devs should care or use it. Adding just a little more info (see my comment) would go a long way IMHO.

**Comment by jeffposnick** [comment]
Thanks, @mathiasbynens! I've included that additional information, and also included a link to your [polyfill](https://github.com/mathiasbynens/CSS.escape).

**Comment by mathiasbynens** [comment]
LGTM :+1:

**Comment by addyosmani** [comment]
LGTM2. Thanks both for fleshing this out!

**Pull request merged**